New article "Let agents inspect w/ Chrome DevTools MCP" (#3811)
MicrosoftDocs/edge-developer: microsoft-edge/devtools/agents/mcp.md

---

> title: Using Microsoft Edge and WebView2 with Chrome DevTools MCP

Reviewer: We can do a little better, and explain what users would want to do this for:

```suggestion
title: Let agents inspect your site and WebView2 app with Chrome DevTools MCP
```

It's a little long, and I'm not 100% happy with it, but the idea is to convey the actual "task" that this allows.
The current title assumes that readers already know what Chrome DevTools MCP is and that they want it.

description: Using Microsoft Edge and WebView2 with Chrome DevTools MCP.
author: MSEdgeTeam
ms.author: msedgedevrel
ms.topic: article
ms.service: microsoft-edge
ms.subservice: devtools
ms.date: 05/27/2026
---
# Using Microsoft Edge and WebView2 with Chrome DevTools MCP

> The [Chrome DevTools MCP](https://github.com/ChromeDevTools/chrome-devtools-mcp) server supports connecting to any Chromium-based browser, including Microsoft Edge and WebView2.  Because the server is built for Chrome, you need to provide extra configuration to point the server at Microsoft Edge or a WebView2 instance.

Reviewer: Before this first line, we should define what Chrome DevTools MCP is, and what it helps developers do.

   * [Step 1: Enable remote debugging for WebView2](#step-1-enable-remote-debugging-for-webview2)
   * [Step 2: Find the WebView2 user data directory](#step-2-find-the-webview2-user-data-directory)
   * [Step 3: Configure the MCP server](#step-3-configure-the-mcp-server)
   * [Step 4: Test your setup](#step-4-test-your-setup)
* [How it works](#how-it-works)
* [Configuring other MCP clients](#configuring-other-mcp-clients)
   * [Copilot CLI](#copilot-cli)
   * [Other MCP clients (Claude Code, Cursor, Gemini CLI, etc.)](#other-mcp-clients-claude-code-cursor-gemini-cli-etc)
* [Troubleshooting](#troubleshooting)
   * ["Could not connect to Chrome" error with auto-connect](#could-not-connect-to-chrome-error-with-auto-connect)
   * [Edge not found at executablePath](#edge-not-found-at-executablepath)
   * [WebView2 won't connect](#webview2-wont-connect)
* [Further reading](#further-reading)

<!-- ====================================================================== -->
## Prerequisites

* [Node.js](https://nodejs.org) LTS version
* [npm](https://www.npmjs.com)
* Microsoft Edge installed (any channel: Stable, Beta, Dev, or Canary)

> * An MCP client such as VS Code (with GitHub Copilot), Copilot CLI, Claude Code, Cursor, etc.

Reviewer: ```suggestion
* A coding agent with MCP support such as VS Code (with GitHub Copilot), Copilot CLI, Claude Code, Cursor, etc.
```

> [!NOTE]

> > The examples in this guide use the VS Code `mcp.json` format.  If you're using a different MCP client, see [Configuring other MCP clients](#configuring-other-mcp-clients) at the end of this guide.

Reviewer: This probably doesn't need to be a formatted NOTE.

<!-- ====================================================================== -->
## 1. Launching Edge

> In this mode, the MCP server launches Edge directly using the `--executablePath` flag pointed at your Edge binary.

Reviewer: ```suggestion
Use this configuration to let your coding agent launch Edge directly for you.

With this configuration, the MCP server, which your agent connects to, launches Edge directly by using the `--executablePath` flag pointing to where the Edge binary is located.
```

Copy and paste the configuration snippet for your platform into your VS Code `mcp.json`.  These examples use Edge Stable.

##### [Windows](#tab/windows/)

```json
{
  "servers": {
    "chrome-devtools": {
      "type": "stdio",
      "command": "npx",
      "args": [
        "-y",
        "chrome-devtools-mcp@latest",
        "--executablePath=%ProgramFiles(x86)%\\Microsoft\\Edge\\Application\\msedge.exe"
      ]
    }
  }
}
```

##### [macOS](#tab/macos/)

```json
{
  "servers": {
    "chrome-devtools": {
      "type": "stdio",
      "command": "npx",
      "args": [
        "-y",
        "chrome-devtools-mcp@latest",
        "--executablePath=/Applications/Microsoft Edge.app/Contents/MacOS/Microsoft Edge"
      ]
    }
  }
}
```

##### [Linux](#tab/linux/)

```json
{
  "servers": {
    "chrome-devtools": {
      "type": "stdio",
      "command": "npx",
      "args": [
        "-y",
        "chrome-devtools-mcp@latest",
        "--executablePath=/usr/bin/microsoft-edge"
      ]
    }
  }
}
```

---

<!-- ====================================================================== -->

> ## Additional flags

Reviewer: Should this be an h4 section below the previous h2 sections?
Or, if we prefer this to be a h2 for visibility, we should rename it to something like: "Additional flags for launching Edge" so it's more directly linked to the previous section.

<!-- ====================================================================== -->
## Edge executable paths for other channels

If you're not using Edge Stable, replace the `--executablePath` value with the appropriate path for your channel.

##### [Windows](#tab/windows/)

| Channel | Default Path |
|---|---|
| Stable | `%ProgramFiles(x86)%\Microsoft\Edge\Application\msedge.exe` |
| Beta | `%ProgramFiles(x86)%\Microsoft\Edge Beta\Application\msedge.exe` |
| Dev | `%ProgramFiles(x86)%\Microsoft\Edge Dev\Application\msedge.exe` |
| Canary | `%LOCALAPPDATA%\Microsoft\Edge SxS\Application\msedge.exe` |

##### [macOS](#tab/macos/)
   
| Channel | Default Path |
|---|---|
| Stable | `/Applications/Microsoft Edge.app/Contents/MacOS/Microsoft Edge` |
| Beta | `/Applications/Microsoft Edge Beta.app/Contents/MacOS/Microsoft Edge` |
| Dev | `/Applications/Microsoft Edge Dev.app/Contents/MacOS/Microsoft Edge` |
| Canary | `/Applications/Microsoft Edge Canary.app/Contents/MacOS/Microsoft Edge` |

##### [Linux](#tab/linux/)

| Channel | Default Path |
|---|---|
| Stable | `/usr/bin/microsoft-edge` |
| Beta | `/usr/bin/microsoft-edge-beta` |
| Dev | `/usr/bin/microsoft-edge-dev` |

---

> [!NOTE]

> > These are default install locations.  Paths may vary based on user configuration, version, or group policies.  Linux does not have a Canary channel.

Reviewer: I would remove the NOTE format here, and move this up to be before the tab set.

<!-- ====================================================================== -->
## 2. Auto-connecting to a running Edge instance

> In this mode, you start Edge yourself (or it's already running), and the MCP server connects to it.  This is useful when:

Reviewer: ```suggestion
Use the configuration below to connect to an already running Edge instance. This is useful when:
```
